### ISSUE TITLE
Cancellation errors and long delays in PUT/POST v0/projects result in 500 responses

### ISSUE THREAD

**Issue by pcrespov** [ITISFoundation/osparc-simcore#1917]
Some e2e tests identify delay and cancellation errors in PUT/POST operations on project resource due to:

- cancellation errors on director API client requests.

See Sleeper failure https://git.speag.com/oSparc/e2e-testing/-/jobs/349797 in 
[https://app.zenhub.com/files/118596920/de843dc1-7f7c-4f16-99cc-873b9d071b5f/download](https://app.zenhub.com/files/118596920/de843dc1-7f7c-4f16-99cc-873b9d071b5f/download)

**Comment by GitHK** [comment]
From past issues with aiohttp, I think we are missing some timeouts on the connections. Which are absolutely necessary necessary to avoid hangs in requests (especially now that we are bouncing connections).

The [aiohttp ClientTimeouts ](https://docs.aiohttp.org/en/stable/client_quickstart.html#aiohttp-client-timeouts) should be of help and [this is the line in packages/service-library/src/servicelib/client_session.py#L2](https://github.com/ITISFoundation/osparc-simcore/blob/bfb7bbbfdca76b19678a465b24642358ebf23c0f/packages/service-library/src/servicelib/client_session.py#L22) where to apply the changes.

Maybe a good timeout should be 1 second (but I know there are requests which take linger). I would also 1 second for `connect` and `sock_connect`. I am not sure how `sock_read` works.
To recap Ideally I will go for:

    ClientTimeout(total=1, connect=1, sock_connect=1)

But I think that the following will work better (meaning all requests must finish in maximum 5 seconds and will fail quickly if the server is not available:

    ClientTimeout(total=5, connect=1, sock_connect=1)

**Comment by GitHK** [comment]
@pcrespov @sanderegg if you agree I will create a PR with this enhancement. My only doubt is the `total` value of the timeout.

**Comment by sanderegg** [comment]
what do you intend to do with calls that take longer?
like getting the list of services takes a while (for sure more than 1 second). the rest seems reasonable

Also I was reading about using HTTP/2 in httpx... would using this not improve speed between the modules?

**Comment by GitHK** [comment]
How would you go about enabling HTTP/2 on the services?

**Comment by GitHK** [comment]
For requests expecting long responses, could it make sense to add a header `X-ExtendTotalTimeout` expressed in seconds to requests which we know might take longer to finish. Meaning that this header might need to be added to some requests from the frontend and other services using this feature.

The proxy in the middle will change the timeout based on the header.

**Comment by GitHK** [comment]
Also we might need to test that httpx can use HTTP/2 without https.

**Comment by pcrespov** [comment]
obsolete
